src/linux/gpu.rs

```diff
@@ -0,0 +1,220 @@
+#![allow(clippy::too_many_arguments)]
+
+use std::fs::{read_dir, File};
+use std::io::Read;
+use std::path::Path;
+
+use crate::GpuExt;
+
+#[doc = include_str!("../../md_doc/gpu.md")]
+pub struct Gpu {
+    name: String,
+    gpu_usage: Option<f32>,
+    vram_used: Option<u64>,
+    vram_total: Option<u64>,
+    freq: Option<u64>,
+    freq_max: Option<u64>,
+    vendor_id: String,
+    brand: Option<String>,
+}
+
+impl Gpu {
+    /// Creates a new instance of `Gpu` with everything set to `None`.
+    pub fn new() -> Self {
+        Gpu {
+            name: String::new(),
+            gpu_usage: None,
+            vram_used: None,
+            vram_total: None,
+            freq: None,
+            freq_max: None,
+            vendor_id: String::new(),
+            brand: None,
+        }
+    }
+
+    /// Find all the GPUs in the system.
+    pub fn get_gpus() -> Vec<Self> {
+        let mut gpus = Vec::new();
+        if let Ok(dir) = read_dir(&Path::new("/sys/class/drm/")) {
+            for entry in dir.flatten() {
+                let entry = entry.path();
+                if entry.is_dir()
+                    && entry
+                        .file_name()
```

Comment:
Instead of calling `entry.file_name().and_then(...)` twice, better store it in a variable.

And to keep the indeed under control, you can do before:

```rust
if !entry.is_dir() {
    continue;
}
```

Reply:
Done in 2f02dc85deda5a09445e1dc47cf58332662ed2fc